ibis/backends/duckdb/__init__.py

```diff
@@ -1556,8 +1556,12 @@ def _register_in_memory_tables(self, expr: ir.Expr) -> None:
             self._register_in_memory_table(memtable)
 
     def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
-        # only register if we haven't already done so
-        if (name := op.name) not in self.list_tables():
+        name = op.name
+        try:
+            # this handles tables _and_ views
+            self.con.table(name)
```

Comment:
This does some extra unnecessary work that it might be nice to avoid (fine to do in a follow-up PR).
- Queries the schema of the table
- Coerces the schema into an ibis schema
- Creates a table object

In the future it might be nice to replace this with a `con.exists`/`con.table_exists`/`con.has_table` API (as proposed in #9215).

Alternatively/additionally we might add a connection-local cache for handling memtables on the `con` object itself. Since memtables have unique ids for names, a simple `dict`/`set` on the `con` should be sufficient for avoiding re-registering tables, and shouldn't run the risk of getting out of sync.

Reply:
Note this isn't calling `self.table`, it's `self.con.table`

Reply:
Ah, missed that. Thanks.